test/extended/clusterup.sh

```diff
@@ -0,0 +1,194 @@
+#!/bin/bash
+source "$(dirname "${BASH_SOURCE}")/../../hack/lib/init.sh"
+trap os::test::junit::reconcile_output EXIT
+
+
+os::util::ensure::built_binary_exists 'oc'
+export PATH="$(dirname "$( os::util::find::built_binary oc )")":"${PATH}"
+
+function os::test::extended::clusterup::run_test () {
+    local test="${1}"
+    local funcname="os::test::extended::clusterup::${test}"
+    local default_home="${HOME}"
+    echo "starting test -- ${test}"
+
+    os::test::junit::declare_suite_start "extended/clusterup/${test}"
+
+    export HOME="$(mktemp -d)"
+    pushd "${HOME}" &> /dev/null
+    echo "Using ${HOME} as home directory"
+    ${funcname}
+    popd &> /dev/null
+    rm -rf "${HOME}"
+    export HOME="${default_home}"
+
+    echo "cluster up -- ${test}: ok"
+    os::test::junit::declare_suite_end
+}
+
+function os::test::extended::clusterup::verify_router_and_registry () {
+    os::cmd::expect_success "oc login -u system:admin"
+    os::cmd::expect_success_and_text "oc get svc -n default" "docker-registry"
+    os::cmd::expect_success_and_text "oc get svc -n default" "router"
+    os::cmd::try_until_success "oc get endpoints docker-registry -n default | grep ':5000'" $(( 10*minute )) 1
+    os::cmd::try_until_success "oc get endpoints router -n default | grep ':80'" $(( 10*minute )) 1
+    os::cmd::expect_success "oc login -u developer"
+}
+
+function os::test::extended::clusterup::verify_image_streams () {
+    os::cmd::try_until_success "oc get is -n openshift ruby -o jsonpath='{ .status.tags[*].tag }' | grep latest" $(( 10*minute )) 1
```

Comment:
You can grab the latest tag with `-o jsonpath='{ .status.tags[?(@.tag=="latest")] }' and maybe test with 

```
os::cmd::try_until_text "oc get is -n openshift ruby -o jsonpath='{ .status.tags[?(@.tag=="latest")].tag }'" "latest" "$(( 10*minute ))" 1
```